Content.Server/Administration/ServerAPI.cs

```diff
@@ -0,0 +1,795 @@
+        await context.RespondAsync(jObject.ToString(), HttpStatusCode.OK);
+        return true;
+    }
+
+    /// <summary>
+    ///    Returns an array containing all game rules.
+    /// </summary>
+    private async Task<bool> GetGameRules(IStatusHandlerContext context)
+    {
+        if (context.RequestMethod != HttpMethod.Get || context.Url!.AbsolutePath != "/admin/game_rules")
+        {
+            return false;
+        }
+
+        if (!CheckAccess(context))
+        {
+            await context.RespondErrorAsync(HttpStatusCode.Unauthorized);
+            return true;
+        }
+
+        var jObject = new JsonObject();
+        var gameRules = new List<string>();
+        foreach (var gameRule in _prototypeManager.EnumeratePrototypes<EntityPrototype>())
+        {
+            if (gameRule.Abstract)
+                continue;
+
+            if (gameRule.HasComponent<GameRuleComponent>(_componentFactory))
+                gameRules.Add(gameRule.ID);
+        }
+
+        jObject["game_rules"] = JsonNode.Parse(JsonSerializer.Serialize(gameRules));
+        await context.RespondAsync(jObject.ToString(), HttpStatusCode.OK);
+        return true;
+    }
+
+
+    /// <summary>
+    ///     Handles fetching information.
+    /// </summary>
+    private async Task<bool> InfoHandler(IStatusHandlerContext context)
+    {
+        if (context.RequestMethod != HttpMethod.Get || context.Url!.AbsolutePath != "/admin/info" || _token == string.Empty)
+        {
+            return false;
+            // 404
+        }
+
+        if (!CheckAccess(context))
+        {
+            await context.RespondErrorAsync(HttpStatusCode.Unauthorized);
+            return true;
+        }
+
+        if (!CheckActor(context, out var actor))
+        {
+            await context.RespondAsync("An actor is required to perform this action.", HttpStatusCode.BadRequest);
+            return true;
+        }
+
+        /*  Information to display
+            Round number
+            Connected players
+            Active admins
+            Active game rules
+            Active game preset
+            Active map
+            MOTD
+            Panic bunker status
+         */
+
+        var ticker = _entitySystemManager.GetEntitySystem<GameTicker>();
+        var adminSystem = _entitySystemManager.GetEntitySystem<AdminSystem>();
+
+        var jObject = new JsonObject();
+
+        jObject["round_id"] = await RunOnMainThread(() => ticker.RoundId);
+
+        var players = new List<Player>();
+        var onlineAdmins = new List<Player>();
+        var onlineAdminsDeadmined = new List<Player>();
+
+        foreach (var player in _playerManager.Sessions)
+        {
+            players.Add(new Player
+            {
+                Guid = player.UserId.UserId.ToString(),
+                Name = player.Name
+            });
+            if (await RunOnMainThread(() => _adminManager.IsAdmin(player)))
```

Comment:
Running `RunOnMainThread()` multiple times like this will make this code take like 3+ seconds on a full server, just because of all the ping-ponging between threads.